client/adjudicate.go

```diff
@@ -444,20 +455,23 @@ func (c *Channel) ensureRegistered(ctx context.Context) error {
 	return nil
 }
 
-// waitForRegisteredEvent waits until a RegisteredEvent has been observed.
-func waitForRegisteredEvent(sub channel.AdjudicatorSubscription, events chan<- *channel.RegisteredEvent) {
-	defer func() {
-		err := sub.Close()
-		if err != nil {
-			log.Warn("Subscription closed with error:", err)
-		}
-	}()
-
-	// Scan for registered event.
+// awaitRegistered scans for an event indicating that the channel has been registered.
+func awaitRegistered(sub channel.AdjudicatorSubscription, events chan<- channel.AdjudicatorEvent) {
```

Comment:
You are ignoring any error that `sub` produces, in case it breaks for example.  
`sub.Err()` should be used after `sub.Next()` returned nil.

Reply:
OK, I refactored the code a bit. Now it also checks `sub.Err` and returns more quickly in the error case.